doc/reference/networking/can_api.rst

```diff
@@ -0,0 +1,335 @@
+  if (filter_id < 0) {
+    LOG_ERR("Unable to attach isr [%d]", filter_id);
+  }
+
+This example shows how to attach a callback from a work-queue.
+In contrast to the `can_attach_isr` function, here the callback is called from the
+work-queue provided. In this case, it is the system work queue. Blocking is
+generally allowed in the callback but could result in a frame backlog when it is
+not limited. For the reason of a backlog, a ring-buffer is applied for every
+attached filter. The size of this buffer can be adjusted in Kconfig.
+This function is not yet callable from userspace context but will be in the
+future.
+
+The filter for this example is configured to match a filter range from
+0x120 to x12f.
+
+.. code-block:: C
+
+  const struct zcan_filter my_filter = {
+          .id_type = CAN_STANDARD_IDENTIFIER,
+          .rtr = CAN_DATAFRAME,
+          .std_id = 0x120,
+          .rtr_mask = 1,
+          .std_id_mask = 0x7F0
+  };
+  struct zcan_work rx_work;
+  int filter_id;
+  struct device *can_dev;
+
+  can_dev = device_get_binding("CAN_0");
+
+  filter_id = can_attach_workq(can_dev, &k_sys_work_q, &rx_work, callback_arg, callback_arg, &my_filter);
+  if (filter_id < 0) {
+    LOG_ERR("Unable to attach isr [%d]", filter_id);
+  }
+
+Here an example for `can_attach_msgq` is shown. With this function, it is
+possible to receive frames synchronously. This function can be called from
+userspace context.
+The size of the message queue should be as big as the expected backlog.
+
+The filter for this example is configured to match the extended identifier
+0x1234567 exactly.
+
+.. code-block:: C
+
+  const struct zcan_filter my_filter = {
+          .id_type = CAN_EXTENDED_IDENTIFIER,
+          .rtr = CAN_DATAFRAME,
+          .std_id = 0x1234567,
+          .rtr_mask = 1,
+          .std_id_mask = CAN_EXT_ID_MASK
+  };
+  CAN_DEFINE_MSGQ(my_can_msgq, 2);
+  struct zcan_frame rx_frame;
+  int filter_id;
+  struct device *can_dev;
+
+  can_dev = device_get_binding("CAN_0");
+
+  filter_id = can_attach_msgq(can_dev, &my_can_msgq, &my_filter);
+  if (filter_id < 0) {
+    LOG_ERR("Unable to attach isr [%d]", filter_id);
+    return;
+  }
+
+  while (true) {
+    k_msgq_get(&my_can_msgq, &rx_frame, K_FOREVER);
+    ... do something with the frame ...
+  }
+
+`can_detach` removes the given filter.
+
+.. code-block:: C
+
+  can_detach(can_dev, filter_id);
+
+SocketCAN
+*********
+
+Zephyr additionally supports SocketCAN, a BSD socket implementation of the
+Zephyr CAN API.
+SocketCAN brings the convenience of the well-known BSD Socket API to
+Controller Area Networks. It is compatible with the Linux SocketCAN
+implementation, where many other high-level CAN projects build on top.
+Note that frames are routed to the network stack instead of passed directly,
+which adds some computation and memory overhead.
+
+Samples
+*******
+
+We have two ready-to-build samples demonstrating use of the Zephyr CAN API
+:ref:`Zephyr CAN sample <can-sample>`,
```

Comment:
since there are only two items, replace the comma with the word "and"

Reply:
Done. Thanks for reviewing it again.